src/Resources/js/modules/combined-meal-service.ts

```diff
@@ -0,0 +1,109 @@
+export class CombinedMealService {
+    public static getCombinedMealDishes($meal: JQuery): Dish[] {
+        let dishes: Dish[] = [];
+        $meal.find('.meal-row').each(function () {
+            const $mealRow = $(this);
+            if (1 === $mealRow.data('combined')) {
+                return;
+            }
+
+            let dish: Dish = {
+                title: $mealRow.find('.title').contents().get(0).nodeValue.trim(),
+                slug: $mealRow.data('slug'),
+                variations: []
+            };
+            $mealRow.find('.variation-row').each(function () {
+                const $dishVarRow = $(this);
+                let dishVariation: DishVariation = {
+                    title: $dishVarRow.find('.text-variation').text().trim(),
+                    slug: $dishVarRow.data('slug')
+                };
+                dish.variations.push(dishVariation);
+            });
+            dishes.push(dish);
+        });
+
+        return dishes;
+    }
+
+    /**
+     * @param $checkbox     Combined Dish Checkbox
+     * @param participantID Participation ID for booked combined meal
+     * @param bookedDishIDs Dish IDs in booked combined meal
+     */
+    public static updateCombinedDish($checkbox: JQuery, participantID: number, bookedDishIDs: string[]) {
```

Comment:
We could also reuse some stuff in updateCombinedDish and updateBookedDishes. There were still some duplicates left.

Reply:
Do you mean replace some code in updateCombinedDish with updateBookedDishes method?

Reply:
I mean, extract the common/duplicated code to a private method, because they do the same partially.

Reply:
I noticed that we can replace some code in updateCombinedDish with updateBookedDishes method. After that we need to see if these methods still contain some common code.

Reply:
Yes, I think so.
You could change it if you want. I'm not sure about the logic. You have a better understanding of the logic here. updateCombinedDish is used in the ParticipationUpdateHandler.
updateBookedDishes (originally the name was updateCombinedDish, I had to rename it to avoid a conflict) is used in the mealIndex.
And now, I think, we have something similar for the guest page. 
I guess it should be always the same thing (changing descripition and attributes) or maybe slightly different.

Can you check, please and change it?

Reply:
Ok, I'll have a look at it.

Reply:
Thanks!